### PR TITLE
nancumsum/nancumprod unit tests

### PR DESCRIPTION
**Pull request by crusaderky**
There are currently no unit tests whatsoever for nancumsum / nancumprod

### CODE REVIEW

File: dask/array/tests/test_reductions.py [hunk 1]
```diff
@@ -102,7 +102,6 @@ def test_reduction_errors():
         x.sum(axis=-3)
 
 
-@pytest.mark.slow
```
Comment (mrocklin) [inline dask/array/tests/test_reductions.py]: Why this change?

Comment (crusaderky) [reply dask/array/tests/test_reductions.py]: It runs in 1.1s on my machine. Not sure what the threshold for "slow" is?

Comment (mrocklin) [reply dask/array/tests/test_reductions.py]: The full dask array test suite (2000 tests (including parameterizations)) runs in around 40s on my machine.  The ten slowest (without the slow decorator) are as follows

```
========================== slowest 10 test durations ===========================
0.76s call     dask/array/tests/test_array_core.py::test_cumulative
0.59s call     dask/array/tests/test_masked.py::test_cumulative
0.48s call     dask/array/tests/test_array_core.py::test_concatenate_unknown_axes
0.35s call     dask/array/tests/test_routines.py::test_flip[shape4-flip-kwargs5]
0.35s call     dask/array/tests/test_routines.py::test_flip[shape4-fliplr-kwargs1]
0.34s call     dask/array/tests/test_routines.py::test_flip[shape4-flipud-kwargs0]
0.34s call     dask/array/tests/test_routines.py::test_flip[shape4-flip-kwargs2]
0.33s call     dask/array/tests/test_rechunk.py::test_rechunk_unknown[x3-chunks3]
0.33s call     dask/array/tests/test_array_core.py::test_store_locks
0.33s call     dask/array/tests/test_creation.py::test_tril_triu
```

Longer tests (like 1s) are fine if they are testing very large pieces of functionality that are likely to break from small changes.  This test doesn't seem to fit that characterization though.  I think that it's probably better to avoid testing it locally and only test it on CI instead.

It also doesn't seem related to this PR.  I usually try to avoid small potentially contentious changes like this because they complicate the process of getting otherwise good work from getting in quickly.
